corehq/apps/accounting/utils/downgrade.py

```diff
@@ -28,3 +40,115 @@ def send_downgrade_notice(invoice, context):
+            "subscription for {}".format(invoice.get_domain())
+        )
+        bcc = [settings.GROWTH_EMAIL]
+
+    context.update({
+        'subscriptions_to_downgrade': subscriptions_to_downgrade
+    })
+    send_html_email_async.delay(
+        subject,
+        invoice.contact_emails,
+        render_to_string('accounting/email/downgrade_warning.html', context),
+        render_to_string('accounting/email/downgrade_warning.txt', context),
+        cc=[settings.ACCOUNTS_EMAIL],
+        bcc=bcc,
+        email_from=get_dimagi_from_email())
+
+
+def _send_overdue_notice(invoice, context):
+    if invoice.is_customer_invoice:
+        bcc = None
+    else:
+        bcc = [settings.GROWTH_EMAIL]
+    send_html_email_async.delay(
+        _('CommCare Billing Statement 30 days Overdue for {}'.format(context['domain_or_account'])),
+        invoice.contact_emails,
+        render_to_string('accounting/email/30_days.html', context),
+        render_to_string('accounting/email/30_days.txt', context),
+        cc=[settings.ACCOUNTS_EMAIL],
+        bcc=bcc,
+        email_from=get_dimagi_from_email())
+
+
+def _apply_downgrade_process(oldest_unpaid_invoice, total, today, subscription=None):
+    from corehq.apps.domain.views.accounting import DomainBillingStatementsView, DomainSubscriptionView
+    from corehq.apps.accounting.views import EnterpriseBillingStatementsView
+
+    context = {
+        'total': format(total, '7.2f'),
+        'date_60': oldest_unpaid_invoice.date_due + datetime.timedelta(days=60),
+        'contact_email': settings.INVOICING_CONTACT_EMAIL
+    }
+    if oldest_unpaid_invoice.is_customer_invoice:
+        domain = oldest_unpaid_invoice.subscriptions.first().subscriber.domain
+        context.update({
+            'statements_url': absolute_reverse(
+                EnterpriseBillingStatementsView.urlname, args=[domain]),
+            'domain_or_account': oldest_unpaid_invoice.account.name
+        })
+    else:
+        domain = subscription.subscriber.domain
+        context.update({
+            'domain': domain,
+            'subscription_url': absolute_reverse(DomainSubscriptionView.urlname,
+                                                 args=[domain]),
+            'statements_url': absolute_reverse(DomainBillingStatementsView.urlname,
+                                               args=[domain]),
+            'domain_or_account': domain
+        })
+
+    days_ago = (today - oldest_unpaid_invoice.date_due).days
+    if days_ago >= 61:
+        if not oldest_unpaid_invoice.is_customer_invoice:  # We do not automatically downgrade customer invoices
+            _downgrade_domain(subscription)
+            _send_downgrade_notice(oldest_unpaid_invoice, context)
+    elif days_ago == 58:
+        _send_downgrade_warning(oldest_unpaid_invoice, context)
+    elif days_ago == 30:
+        _send_overdue_notice(oldest_unpaid_invoice, context)
+
+
+def downgrade_eligible_domains():
```

Comment:
What do you think of moving this to the top of the module? It appears to be the most important function in here.